plugins/modules/aci_vmm_controller.py

```diff
@@ -0,0 +1,374 @@
+  returned: info
+  type: list
+  sample:
+    {
+        "fvTenant": {
+            "attributes": {
+                "descr": "Production environment"
+            }
+        }
+    }
+previous:
+  description: The original configuration from the APIC before the module has started
+  returned: info
+  type: list
+  sample:
+    [
+        {
+            "fvTenant": {
+                "attributes": {
+                    "descr": "Production",
+                    "dn": "uni/tn-production",
+                    "name": "production",
+                    "nameAlias": "",
+                    "ownerKey": "",
+                    "ownerTag": ""
+                }
+            }
+        }
+    ]
+proposed:
+  description: The assembled configuration from the user-provided parameters
+  returned: info
+  type: dict
+  sample:
+    {
+        "fvTenant": {
+            "attributes": {
+                "descr": "Production environment",
+                "name": "production"
+            }
+        }
+    }
+filter_string:
+  description: The filter string used for the request
+  returned: failure or debug
+  type: str
+  sample: ?rsp-prop-include=config-only
+method:
+  description: The HTTP method used for the request to the APIC
+  returned: failure or debug
+  type: str
+  sample: POST
+response:
+  description: The HTTP response from the APIC
+  returned: failure or debug
+  type: str
+  sample: OK (30 bytes)
+status:
+  description: The HTTP status from the APIC
+  returned: failure or debug
+  type: int
+  sample: 200
+url:
+  description: The HTTP url used for the request to the APIC
+  returned: failure or debug
+  type: str
+  sample: https://10.11.12.13/api/mo/uni/tn-production.json
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.aci.plugins.module_utils.aci import ACIModule, aci_argument_spec
+
+VM_PROVIDER_MAPPING = dict(
+    cloudfoundry='CloudFoundry',
+    kubernetes='Kubernetes',
+    microsoft='Microsoft',
+    openshift='OpenShift',
+    openstack='OpenStack',
+    redhat='Redhat',
+    vmware='VMware',
+)
+
+VM_SCOPE_MAPPING = dict(
+    cloudfoundry='cloudfoundry',
+    kubernetes='kubernetes',
+    microsoft='MicrosoftSCVMM',
+    openshift='openshift',
+    openstack='openstack',
+    redhat='rhev',
+    vmware='vm',
+)
+
+
+def main():
+    argument_spec = aci_argument_spec()
+    argument_spec.update(
+        name=dict(type='str'),
+        controller_hostname=dict(type='str'),
+        dvs_version=dict(type='str', choices=['unmanaged', '5.1', '5.5', '6.0', '6.5', '6.6', '7.0']),
+        stats_collection=dict(type='str', default='disabled', choices=['enabled', 'disabled']),
+        domain=dict(type='str', aliases=['domain_name', 'domain_profile']),
+        state=dict(type='str', default='present', choices=['absent', 'present', 'query']),
+        credentials=dict(type='str'),
+        management_epg=dict(type='str'),
```

Comment:
as the current management_epg only supports inband, I would either rename it to inband_management_epg or change it to a dictionary so in future we can add the type to it.

Reply:
Or we could directly add support for out of band.